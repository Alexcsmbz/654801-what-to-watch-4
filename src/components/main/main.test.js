import Main from './main.jsx';
import {BrowserRouter as Router} from 'react-router-dom';

const name = `Whatever movie`;
const genre = `Criminal`;
const releaseDate = `1999`;
const movies = [
  {
    name: `movie 1`,
    img: `img/img-1.jpg`,
  },
  {
    name: `movie 2`,
    img: `img/img-2.jpg`,
  },
  {
    name: `movie 3`,
    img: `img/img-3.jpg`,
  },
  {
    name: `movie 4`,
    img: `img/img-4.jpg`,
  },
];
const genres = [
  `Comedy`,
  `Drama`,
  `Crime`,
];

describe(`Main snapshot test`, () => {
  it(`Main should render movies name, genre, relise date, movies list`, () => {
    const tree = renderer.create(
        <Router>
          <Main
            genres={genres}
            movieName={name}
            genre={genre}
            releaseDate={releaseDate}
            movies={movies}
            activeIdx={0}
            setActiveIdx={() => {}}
          />
        </Router>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
