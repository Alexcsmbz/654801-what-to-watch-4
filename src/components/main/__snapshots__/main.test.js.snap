// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main snapshot test Main should render movies name, genre, relise date, movies list 1`] = `
Array [
  <div
    className="visually-hidden"
  >
    <video
      controls={true}
      height="175"
      poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
      width="280"
    >
      <source
        src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
        type="video/webm"
      />
      <source
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        type="video/mp4"
      />
    </video>
  </div>,
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link undefined"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height="327"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Whatever movie
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Criminal
            </span>
            <span
              className="movie-card__year"
            >
              1999
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn--play btn movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn--list btn movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active "
          onClick={[Function]}
        >
          <div
            className="catalog__genres-link"
          >
            Comedy
          </div>
        </li>
        <li
          className="catalog__genres-item undefined"
          onClick={[Function]}
        >
          <div
            className="catalog__genres-link"
          >
            Drama
          </div>
        </li>
        <li
          className="catalog__genres-item undefined"
          onClick={[Function]}
        >
          <div
            className="catalog__genres-link"
          >
            Crime
          </div>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
        >
          <video
            controls={true}
            height="175"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            width="280"
          >
            <source
              type="video/webm"
            />
            <source
              type="video/mp4"
            />
          </video>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/movie-page/movie-1"
              onClick={[Function]}
            >
              movie 1
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
        >
          <video
            controls={true}
            height="175"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            width="280"
          >
            <source
              type="video/webm"
            />
            <source
              type="video/mp4"
            />
          </video>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/movie-page/movie-2"
              onClick={[Function]}
            >
              movie 2
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
        >
          <video
            controls={true}
            height="175"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            width="280"
          >
            <source
              type="video/webm"
            />
            <source
              type="video/mp4"
            />
          </video>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/movie-page/movie-3"
              onClick={[Function]}
            >
              movie 3
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
        >
          <video
            controls={true}
            height="175"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            width="280"
          >
            <source
              type="video/webm"
            />
            <source
              type="video/mp4"
            />
          </video>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/movie-page/movie-4"
              onClick={[Function]}
            >
              movie 4
            </a>
          </h3>
        </article>
      </div>
      <div
        className="catalog__more"
      />
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
